include/seastar/http/httpd.hh

```diff
@@ -171,6 +172,10 @@ public:
     [[deprecated("use listen(socket_address addr, server_credentials_ptr credentials)")]]
     void set_tls_credentials(server_credentials_ptr credentials);
 
+    void set_keepalive_parameters(std::optional<net::keepalive_params> params) {
```

Comment:
I wonder if anybody would ever want to call this function with a disengaged optional. Wouldn't you only call it if you want to set the parameters? Maybe the parameter of this function shouldn't be an optional?
(it's not really wrong that it's optional, I just find it strange).

Reply:
I have no strong feelings either way, I think most people just set it once and forget about it but as Travis suggested in another comment one could imagine that one might want to reset it.